### PR TITLE
Snippets view improvements

### PR DESCRIPTION
**Pull request by magland**

### CODE REVIEW

File: python/labbox_ephys/helpers/prepare_snippets_h5.py [hunk 1]
```diff
@@ -6,15 +6,15 @@
...
+@hi.function('prepare_snippets_h5', '0.2.6')
 @hi.container('docker://magland/labbox-ephys-processing:0.3.19')
 @hi.local_modules(['../../labbox_ephys'])
 def prepare_snippets_h5(
     recording_object,
     sorting_object,
     start_frame=None,
     end_frame=None,
-    max_events_per_unit=1000,
+    max_events_per_unit=None,
```
Comment (magland) [inline python/labbox_ephys/helpers/prepare_snippets_h5.py:17]: The default max_events_per_unit is now None, meaning that we include all events in the snippets file by default.

File: src/extensions/averagewaveforms/AverageWaveformsView/WaveformWidget.tsx [hunk 2]
```diff
@@ -15,13 +15,17 @@ export type Props = {
     height: number
     selection: RecordingSelection
     selectionDispatch: RecordingSelectionDispatch
+    electrodeOpts: ElectrodeOpts
+}
+
+export type ElectrodeOpts = {
+    colors?: ElectrodeColors
+    showLabels?: boolean,
+    disableSelection?: boolean
```
Comment (magland) [inline src/extensions/averagewaveforms/AverageWaveformsView/WaveformWidget.tsx:24]: Option to disable selection of electrodes in the waveform widget.

File: src/extensions/snippets/SnippetsView/SnippetsRow.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
+import { GridList, GridListTile } from '@material-ui/core';
```
Comment (magland) [inline src/extensions/snippets/SnippetsView/SnippetsRow.tsx:1]: Renamed SnippetsWidgetBox to SnippetsRow -- better name for what it is -- a row of snippets for a single unit.

File: src/extensions/snippets/SnippetsView/SnippetsWidget.tsx [hunk 4]
```diff
@@ -13,18 +14,34 @@ type Props = {
     height: number
 }
 
-const SnippetsWidget: FunctionComponent<Props> = ({ recording, sorting, selection, selectionDispatch, unitIds, width }) => {
+const WhenVisible: FunctionComponent<{width: number, height: number}> = ({width, height, children}) => {
```
Comment (magland) [inline src/extensions/snippets/SnippetsView/SnippetsWidget.tsx:17]: This WhenVisible is a convenient wrapper component that hides its child until it has come visible (via scrolling, eg) for the first time.

File: src/extensions/snippets/SnippetsView/SnippetsRow.tsx (hunk 3 above)
Comment (jsoules) [inline src/extensions/snippets/SnippetsView/SnippetsRow.tsx:1]: This is pretty much replacing the `SnippetsWidgetBox` class? The code all looks really familiar from the last PR.

**Review by jsoules — APPROVED** [review]

File: src/extensions/snippets/SnippetsView/SnippetsRow.tsx (hunk 3 above)
Comment (magland) [reply src/extensions/snippets/SnippetsView/SnippetsRow.tsx:1]: Yes, mostly a rename

**Pull request merged**